tests/unit/client.py

```diff
@@ -0,0 +1,659 @@
+# Copyright 2013-2014 Massachusetts Open Cloud Contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the
+# License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS
+# IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied.  See the License for the specific language
+# governing permissions and limitations under the License.
+
+"""Unit tests for client library"""
+import haas
+from haas import model, api, deferred, server, config
+from haas.model import db
+from haas.network_allocator import get_network_allocator
+import pytest
+import json
+import requests
+import os
+import tempfile
+import subprocess
+import time
+from subprocess import check_call, Popen
+from urlparse import urljoin
+import requests
+from requests.exceptions import ConnectionError
+from haas.client.base import ClientBase
+from haas.client.client import Client, RequestsHTTPClient, KeystoneHTTPClient
+
+
+ep = "http://127.0.0.1:8000" or os.environ.get('HAAS_ENDPOINT')
+username = "hil_user" or os.environ.get('HAAS_USERNAME')
+password = "hil_pass1234" or os.environ.get('HAAS_PASSWORD')
+
+http_client = RequestsHTTPClient()
+http_client.auth = (username, password)
+C = Client(ep, http_client)  # Initializing client library
+MOCK_SWITCH_TYPE = 'http://schema.massopencloud.org/haas/v0/switches/mock'
+OBM_TYPE_MOCK = 'http://schema.massopencloud.org/haas/v0/obm/mock'
+OBM_TYPE_IPMI = 'http://schema.massopencloud.org/haas/v0/obm/ipmi'
+
+
+class Test_ClientBase:
+    """ When the username, password is not defined
+    It should raise a LookupError
+    """
+
+    def test_init_error(self):
+        try:
+            x = ClientBase()
+        except TypeError:
+            assert True
+
+# FIX ME: The test may vary based on which backend session is used.
+#    def test_correct_init(self):
+#        x = ClientBase(ep, 'some_base64_string')
+#        assert x.endpoint == "http://127.0.0.1:8000"
+#        assert x.auth == "some_base64_string"
```

Comment:
Don't leave code commented out. This looks like it was testing something that's obviated by the change to take an http client, so we should just delete it.

For reference, if you have a test that is failing, fixing it is preferred, but if there's some reason you're not going to (e.g. the test is for a discovered bug that hasn't been sorted out yet), you can use `pytest.mark.xfail`.